packages/confetti-ds/src/IconButton.tsx

```
@@ -0,0 +1,56 @@
+import React, { SyntheticEvent } from "react";
+import { isUndefined } from "lodash";
+
+import Icon from "./Icon";
+import { IconTypes } from "./constants";
+
+export interface IconButtonProps {
+  /** Sets the icon related to the button label. Default state: no icon. */
+  icon: IconTypes;
+  /** Icon Button theme variation. */
+  theme?: "teal" | "purple" | "light" | "dark";
+  /** Disables the Button. Will be read by screen readers. When true, will override `disabled`. */
+  ariaDisabled?: boolean;
+  /** Disables the Icon Button. Won't be read by screen readers. */
+  disabled?: boolean;
+  /** Action to be executed when the button is clicked. */
+  onClick?: (event: SyntheticEvent) => any;
+}
+
+export default function IconButton({
+  icon,
+  theme = "teal",
+  ariaDisabled = false,
+  disabled = false,
+  onClick = (event) => {},
+}: IconButtonProps) {
+  const renderIcon = () =>
+    icon ? (
```

quick question: does it make sense for an IconButton to not render an icon?
because it's marked as required, but we're rendering it empty if people don't pass it.
maybe it would make more sense to just render it and let it error out instead of only relying on the type to check it.